Methane_MarkDown.Rmd

```diff
@@ -111,3 +110,206 @@ ggplot(data= results, aes(x= year, y= value, color= units))+
+  output[[rownum]]$run_number<- data$run_number[[rownum]] #now that row has finished without 5th row, put 5th row back in as guide for next row run
+}
+
+output_data<- output %>% bind_rows() #create a data frame with output
+full_data<- left_join(output_data, data, by= "run_number") #add in param values columns to results of loop df by pairing with row_number column
+```
+
+### Graphing multiple parameters
+
+Now that we've created multi-parameter data, we can visualize our results. Using the `pivot_longer` function from `tidyr` we first group our parameters into two columns: `parameter` and `param_value`. This will make graphing much easier, since we are now only working with one variable type.
+
+```{r piv-long-data}
+long_fdata<- full_data %>%
+  pivot_longer(!c(scenario, year, variable, value, units, run_number), names_to= "parameter", values_to= "param_value")
+```
+
+To visualize the spread of our new values we can create a histogram broken into four windows with `facet_wrap`.
+
+```{r long-hist}
+long_fdata %>%
+  ggplot(aes(x= param_value, fill= parameter))+
+  geom_histogram(alpha= 0.6, binwidth= 7)  +
+   facet_wrap(~parameter, scales = "free_y")
+```
+To show the differences in values between each run, we can create a bar chart grouping parameter by `run_number`. Although the values for each run stay consistent, we must use the `summarise` and `mean` functions to condense our data frame into single values before graphing. 
+
+```{r bar_params}
+mean_fdata<- full_data %>%
+  group_by(run_number) %>%
+  summarise(meanNatM_TgCH4= mean(NATURAL_CH4), meanPreM_ppbvCH4= mean(PREINDUSTRIAL_CH4), meanSoilM_Years= mean(LIFETIME_SOIL), meanStratM_Years= mean(LIFETIME_STRAT))
+
+long_mdata<- mean_fdata %>%
+  pivot_longer(!run_number, names_to= "parameter", values_to= "param_value")
+
+ggplot(long_mdata, aes(y= param_value, x= run_number , fill= parameter )) +
+  geom_bar(stat="identity",
+           position= position_dodge())
+```
+
+We can also create a line graph for our output values. This is the same as the sensitivity analysis graph earlier on, only now with more input parameters. This graph isn't ideal if we are looping more than a handful of times...
+
+```{r multi_sensitivity}
+ggplot(full_data, aes(y= value, x=  year, color= run_number, group = run_number)) +
+  geom_line() +
+  facet_wrap(~variable, scales = "free_y")+
+  scale_color_viridis_c() 
+#shutdown(core)
+```
+
+... A better graph for our runs is `geom_line`, which graphs the mean of our outputs each year, as well as the maximum and minimum values. First we find the mean, standard deviation, minimum and maximum using `summarize` - then we can plot the results.
+
+```{r summary-graph}
+
+sum_data<-  long_fdata%>%
+  group_by(variable, year) %>%
+  summarize(Mean_val= mean(value), SD_val= sd(value), Min_val= min(value), Max_val= max(value))
+
+ggplot(sum_data, aes(year)) +
+   geom_ribbon(aes(ymin= Min_val, ymax= Max_val), fill= "pink") +
```

Comment:
Try: `geom_ribbon(aes(ymin= Min_val, ymax= Max_val, fill= variable), alpha = 0.3)`
`fill` is passed in as an aesthetic because it draws from your data, `alpha` is just you setting a value arbitrarily and is therefore not an aesthetic.